lib/paranoia.rb

```diff
@@ -205,56 +199,102 @@ def restore_associated_records
   end
 end
 
+class Class
+  def extend?(klass)
+    not superclass.nil? and ( superclass == klass or superclass.extend? klass )
+  end
+end
+
 class ActiveRecord::Base
-  def self.acts_as_paranoid(options={})
+  class << self
+    alias_method :original_connection, :connection
```

Comment:
Ruby 2.0 allows you to prepend modules and then call super in the methods to call the overridden methods.
http://gshutler.com/2013/04/ruby-2-module-prepend/